meson.build

```diff
@@ -17,7 +17,11 @@ shellcheck = find_program('shellcheck', required: false)
 profiledir = get_option('profile_dir')
 
 systemd_dep = dependency('systemd')
```

Comment:
Like you suspected, `systemd_dep` can moved inside the `if` branch because it's otherwise not used.